src/OrleansSQLUtils/Storage/RelationalStorage.cs

```diff
@@ -146,11 +146,7 @@ public Task<IEnumerable<TResult>> ReadAsync<TResult>(string query, Action<IDbCom
                 throw new ArgumentNullException("selector");
```

Comment:
@jdom Statistics tests fail to `AggregateException` and the inner exception is `NotImplementedException`. Statistics is implemented, so this is probably something else with the setup (that could be vastly simplified, I have code to do constructor injection in XUnit.NET, for instance). ![Statistics tests fail](https://user-images.githubusercontent.com/5158694/27768067-b03bb6da-5f12-11e7-973f-c7f4070e1b75.png)

The second set of tests that fail is persistence tests, but here again there isn't a clear reason other than that inconsistent write is reported whereas there should be. Looking at the tests, these shouldn't have anything special in way of data, but they use `Func<IInternalGrainFactory, GrainReference> getGrainReference` which is different from 
```
internal Tuple<GrainReference, GrainState<TestState1>> GetTestReferenceAndState(long grainId, string version)
{
    return Tuple.Create(this.grainFactory.GetGrain(GrainId.GetGrainId(UniqueKey.NewKey(grainId, UniqueKey.Category.Grain))), new GrainState<TestState1> { State = new TestState1(), ETag = version });
}
```
to generate a reference. It feels like grain reference shouldn't be the case to `InconsistentStateException` as such, but I wonder if the case could be such that on write-read cycles a wrong reference is combined with a state and hence a subsequent update cycle won't locate a reference in the storage, fails to update it and hence an exception is reported.
![InconsistentStateException](https://user-images.githubusercontent.com/5158694/27768094-c3a62984-5f13-11e7-9b99-ba876776a78c.png)

I write these as a head's up since I'm on the road today, so to speak, and likely won't have many minutes to put time besides this. I'm not sure about the following week (I have little time at least).